cupy/functional/piecewise.py

```diff
@@ -0,0 +1,48 @@
+import cupy
+
+
+def piecewise(x, condlist, funclist):
+    """Evaluate a piecewise-defined function.
+
+        Args:
+            x (cupy.ndarray): input domain
+            condlist (list of cupy.ndarray or bool scalars):
+                Each boolean array/ scalar corresponds to a function
+                in funclist. Length of funclist is equal to that of
+                condlist. If one extra function is given, it is used
+                as the default value when the otherwise condition is met
+            funclist (cupy.ndarray or list): list of scalar functions.
+
+        Returns:
+            cupy.ndarray: the scalar values in funclist on portions of x
+                defined by condlist.
+
+        .. warning::
+
+            This function currently doesn't support callable functions,
+            args and kw parameters.
+
+        .. seealso:: :func:`numpy.piecewise`
+        """
+
+    if any(callable(item) for item in funclist):
+        raise NotImplementedError(
+            'Callable functions are not supported currently')
```

Comment:
How about moving this check into the for-loop [here](https://github.com/cupy/cupy/pull/3329/files#diff-26cc69efdac81a9dee1d8f0ebc5d0356R46-R47)?